src/mapml/layers/FeatureIndexOverlay.js

```diff
@@ -0,0 +1,170 @@
+        this._body.index = 0;
+
+        map.on("layerchange layeradd layerremove overlayremove", this._toggleEvents, this);
+        map.on('moveend focus', this._checkOverlap, this);
+        map.on("keydown", this._onKeyDown, this);
+        this._addOrRemoveFeatureIndex();
+    },
+
+    _checkOverlap: function () {
+        this._map.fire("mapkeyboardfocused");
+        let bounds = this._map.getPixelBounds();
+        let center = bounds.getCenter();
+        let wRatio = Math.abs(bounds.min.x - bounds.max.x) / (this._map.options.mapEl.width);
+        let hRatio = Math.abs(bounds.min.y - bounds.max.y) / (this._map.options.mapEl.height);
+
+        let w = wRatio * (getComputedStyle(this._container).width).replace(/\D/g,'') / 2;
+        let h = hRatio * (getComputedStyle(this._container).height).replace(/\D/g,'') / 2;
+        let minPoint = L.point(center.x - w, center.y + h);
+        let maxPoint = L.point(center.x + w, center.y - h);
+        let b = L.bounds(minPoint, maxPoint);
+        let featureIndexBounds = M.pixelToPCRSBounds(b,this._map.getZoom(),this._map.options.projection);
+
+        let features = this._map.featureIndex.inBoundFeatures;
+        let index = 1;
+        let keys = Object.keys(features);
+        let body = this._body;
+
+        body.innerHTML = "";
+        body.index = 0;
+
+        body.allFeatures = [];
+        keys.forEach(i => {
+            let layers = features[i].layer._layers;
+            let bounds = L.bounds();
+
+            if(layers) {
+                let keys = Object.keys(layers);
+                keys.forEach(j => {
+                    if(!bounds) bounds = L.bounds(layers[j]._bounds.min, layers[j]._bounds.max);
+                    bounds.extend(layers[j]._bounds.min);
+                    bounds.extend(layers[j]._bounds.max);
+                });
+            } else if(features[i].layer._bounds){
+                bounds = L.bounds(features[i].layer._bounds.min, features[i].layer._bounds.max);
+            }
+
+            if(featureIndexBounds.overlaps(bounds)){
+                let group = features[i].path;
+                let label = group.getAttribute("aria-label");
+
+                if (index < 8){
+                    body.appendChild(this._updateOutput(label, index, index));
+                }
+                if (index % 7 === 0 || index === 1) {
+                    body.allFeatures.push([]);
+                }
+                body.allFeatures[Math.floor((index - 1) / 7)].push({label, index, group});
+                if (body.allFeatures[1] && body.allFeatures[1].length === 1){
+                    body.appendChild(this._updateOutput("More results", 0, 9));
+                }
+                index += 1;
+            }
+        });
+        this._addToggleKeys();
+    },
+
+    _updateOutput: function (label, index, key) {
+        let span = document.createElement("span");
+        span.setAttribute("index", index);
```

Comment:
To be on the safe side, the `index` attribute should probably be [`data-`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/data-*) prefixed (related issue: https://github.com/Maps4HTML/Web-Map-Custom-Element/issues/191).